packages/fxa-auth-server/lib/routes/account.js

```diff
@@ -1536,9 +1540,13 @@ module.exports = (
 
         await customs.check(request, uid, 'updateEcosystemAnonId');
 
+        if (!ifNoneMatch && !ifMatch) {
+          throw error.anonIdNoCondition()
+        }
+
         const { ecosystemAnonId: existingAnonId } = await db.account();
 
-        if ((ifNoneMatch || ifMatch) && existingAnonId) {
+        if (existingAnonId) {
```

Comment:
If there's no existing anon_id, but the client specifies "if-match: x", should we do the update?

Reply:
Per Slack convo, noted and may be good as a followup.